python/pyarrow/parquet.py

```diff
@@ -807,6 +807,80 @@ def write_table(table, where, row_group_size=None, version='1.0',
         writer.close()
 
 
+def write_to_dataset(table, where, partition_cols, root_path,
+                     preserve_index=True, **kwargs):
+    """
+    Wrapper around parquet.write_table for writing a Table to
+    Parquet format by partitions.
+    For each combination of partition columns and values,
+    a subdirectories are created in the following
+    manner:
+
+    root_dir/
+    ├── group1=value1
+    │   ├── group2=value1
+    │   │   └── <result.parquet>
+    │   └── group2=value2
+    │       └── <result.parquet>
+    ├── group1=value2
+    │   └── group2=value1
+    │       └── <result.parquet>
+    ...
+    └── group1=valueN
+        ├── group2=value1
+        │   └── <result.parquet>
+        ...
+        └── group2=valueN
+            └──<result.parquet>
+
+    Parameters
+    ----------
+    table : pyarrow.Table
+    where: string,
+        Name of the parquet file for data saved in each partition
```

Comment:
In these cases, usually the data files are randomly generated. Any reason not to use a uuid value? (see `guid()` function in `pyarrow.compat`)

Reply:
You could make the file path an optional parameter, so that if it's None a uuid is used

Reply:
Sounds good, I'll take a look at `guid()`. I was also looking at `uuid4()` from the `uuid` module, but glad to know there's already a native function.